corehq/apps/hqwebapp/models.py

```diff
@@ -47,11 +66,20 @@ def save(self, *args, **kwargs):
         super(MaintenanceAlert, self).save(*args, **kwargs)
 
     @classmethod
-    @quickcache([], timeout=60 * 60)
+    @quickcache([], timeout=1 * 60)
     def get_active_alerts(cls):
+        cls._update_active_alerts()
```

Comment:
Okay, that makes sense. It's less intuitive to me to think of it that way, but it definitely feels like it would be more efficient.

Reply:
Yeah, I think it's a safer way to go, since you also wouldn't have to worry about a task failure - just a simpler system.  I actually ran across [another ticket](https://dimagi-dev.atlassian.net/browse/USH-2402) this morning with the same idea and updated the description there to recommend modifying the query rather than updating the model.

Tangentially, I really like the pattern of [modifying the default queryset](https://github.com/dimagi/commcare-hq/blob/5339fb0b41ae09582e3e9e51a349d4cba7c79206/corehq/apps/users/models.py#L2924-L2925) as that model does, which is a more robust way to centralize that logic.  In this case, the model is only really queried through this `get_active_alerts` classmethod, so I think it's fine to put the logic there, but if queries were coming from a bunch of different places, it's nice to have it central.